## chi square test 
 if the variables are dependent upon each other or not it checks the relationship between 2 discrete variables

## ho=they are independent-->no relationship, h1=they are dependent -->relationship

In [1]:
import pandas as pd

In [2]:
d1={'Satisfaction':['Very Satisfied','Satisfied','Neutral','Unsatisfied','Very Unsatisfied'],
    'Smart Thermostat':[50,80,60,30,20],
    'Smart Light':[70,100,90,50,50,]
    }

In [3]:
df=pd.DataFrame(d1)

In [4]:
df

Satisfaction  Smart Thermostat  Smart Light
0    Very Satisfied                50           70
1         Satisfied                80          100
2           Neutral                60           90
3       Unsatisfied                30           50
4  Very Unsatisfied                20           50

In [15]:
import scipy.stats as stats

# first we determine the chi table value then we compare the chitable critical value with the chi test statistic value and based on that we reject the null hypothesis or alternate hypothesis.

In [16]:
critical_chi_table_val=stats.chi2.ppf(q=0.95,df=4)
critical_chi_table_val

9.487729036781154

we got the critical value as 9.48
If our calculated chi-square statistic > 9.488, we reject the null hypothesis,i.e we accept alternate hypothesis .
If our calculated chi-square statistic ≤ 9.488, we fail to reject the null hypothesis, i,e we accept null hypothesis .

In [17]:
df1=df[["Smart Thermostat","Smart Light"]]

In [18]:
chi2, p, dof, expected = stats.chi2_contingency(df1)


In [31]:
print("\033[1mThe Chi-Square Value with degrees of freedom\033[0m",dof,"\033[1mis\033[0m",chi2.round(2))
print("\033[1mThe Pvalue is\033[0m",p.round(2))

The Chi-Square Value with degrees of freedom 4 is 5.64
The Pvalue is 0.23


In [21]:
alpha=0.05
pvalue=0.23

In [33]:
if pvalue<alpha:
    print("\033[1mho is rejected and h1 accepted\033[0m")
else:
    print("\033[1mh1 rejected ho accepted\033[0m")

h1 rejected ho accepted


## similarly

In [25]:
chivalue=5.64
critical_value=9.84

In [34]:
if chivalue>critical_value:
    print("\033[1mho is rejected and h1 accepted\033[0m")
else:
    print("\033[1mh1 rejected ho accepted\033[0m")

h1 rejected ho accepted


# CONCLUSION
## Hence we prove that there is no significant relationship between the Smart Light vs Smart Thermostat ,which also means that there is no relationship between the customer satisfaction level based on it.